CRM/Civicase/XMLProcessor/Report.php

```diff
@@ -0,0 +1,319 @@
+<?php
+class CRM_Civicase_XMLProcessor_Report extends CRM_Case_XMLProcessor_Report {
+  /**
+   * @param int $clientID
+   * @param int $caseID
+   * @param $activityTypes
+   * @param $activities
+   * @param $selectedActivities
+   */
+  public function getActivities($clientID, $caseID, $activityTypes, &$activities, $selectedActivities = null) {
+
+    // get all activities for this case that in this activityTypes set
+    foreach ($activityTypes as $aType) {
+      $map[$aType['id']] = $aType;
+    }
+
+    // get all core activities
+    $coreActivityTypes = CRM_Case_PseudoConstant::caseActivityType(FALSE, TRUE);
+
+    foreach ($coreActivityTypes as $aType) {
+      $map[$aType['id']] = $aType;
+    }
+
+    $activityTypeIDs = implode(',', array_keys($map));
+    $query = "
+SELECT a.*, c.id as caseID
+FROM   civicrm_activity a,
+       civicrm_case     c,
+       civicrm_case_activity ac
+WHERE  a.is_current_revision = 1
+AND    a.is_deleted =0
+AND    a.activity_type_id IN ( $activityTypeIDs )
+AND    c.id = ac.case_id
+AND    a.id = ac.activity_id
+AND    ac.case_id = %1
+";
+
+    $params = array(1 => array($caseID, 'Integer'));
+    $dao = CRM_Core_DAO::executeQuery($query, $params);
+    while ($dao->fetch()) {
+      $activityTypeInfo = $map[$dao->activity_type_id];
+      if (count($selectedActivities) === 0 || in_array($dao->id, $selectedActivities)) {
+        $activities[] = $this->getActivity($clientID,
+          $dao,
+          $activityTypeInfo
+        );
+      }
+    }
+  }
+
+  /**
+   * @param int $clientID
+   * @param int $caseID
+   * @param string $activitySetName
+   * @param array $params
+   * @param CRM_Core_Form $form
+   * @param array $selectedActivities
+   *
+   * @return mixed
+   */
+  public static function getCaseReport($clientID, $caseID, $activitySetName, $params, $form, $selectedActivities = null) {
+
+    $template = CRM_Core_Smarty::singleton();
+
+    $template->assign('caseId', $caseID);
+    $template->assign('clientID', $clientID);
+    $template->assign('activitySetName', $activitySetName);
+
+    if (!empty($params['is_redact'])) {
+      $form->_isRedact = TRUE;
+      $template->assign('_isRedact', 'true');
+    }
+    else {
+      $form->_isRedact = FALSE;
+      $template->assign('_isRedact', 'false');
+    }
+
+    // first get all case information
+    $case = $form->caseInfo($clientID, $caseID);
+    $template->assign_by_ref('case', $case);
```

Comment:
This function is overriding its parent. So all this logic is copied from the parent, so I dont want to spend time fixing the logic, because it would increase the scope of the ticket. But I would fix any code organisation related comments.